compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1033,6 +1034,17 @@ impl<'hir> LoweringContext<'_, 'hir> {
         )
     }
 
+    /// Desugar `<expr>.use` into:
+    /// ```ignore (pseudo-rust)
+    /// <expr>.clone()
+    /// ```
+    fn lower_expr_use(&mut self, use_kw_span: Span, expr: &Expr) -> hir::ExprKind<'hir> {
+        let expr = self.lower_expr(expr);
+        let span = self.mark_span_with_reason(DesugaringKind::Use, use_kw_span, None);
```

Comment:
I don't think we want this `DesugaringKind` anymore

Reply:
I was taking a quick look at this, this is used for diagnostic purposes, see for instance https://github.com/spastorino/rust/blob/b82e779259706ceb63fbd38fa5c9d8b9b632149a/compiler/rustc_borrowck/src/diagnostics/mod.rs#L1200-L1207
Do you meant to say that `.use` doesn't move so this error is not important. Also my code doesn't completely handles the DesugaringKind, so or I should remove if this error shouldn't happen or I should do the actual handling of it :).